CARET_trace/src/hooked_trace_points.cpp

```diff
@@ -254,6 +283,19 @@ void _ZN8eprosima8fastrtps4rtps13WriterHistory13set_fragmentsEPNS1_13CacheChange
   }
 }
 
+// for rmw_cyclonedds_cpp
+// store message address in tread local memory
+rmw_ret_t rmw_publish_serialized_message(
+  const rmw_publisher_t * publisher, const rmw_serialized_message_t * serialized_message,
+  rmw_publisher_allocation_t * allocation)
+{
+  serialized_message_addr = static_cast<const void *>(serialized_message);
+  using functionT = int (*)(const void *, const void *, const void *);
+  static void * orig_func = dlsym(RTLD_NEXT, __func__);
+  int dds_return = ((functionT)orig_func)(publisher, serialized_message, allocation);
+  return dds_return;
```

Comment:
I suggenst changing the type of return value bacause it is not obvious that sizeof(int) is equal to sizeof(rmw_ret_t). Also, I suggest changing variable name because it is not a dds function.

```suggestion
  using functionT = rmw_ret_t (*)(const void *, const void *, const void *);
  static void * orig_func = dlsym(RTLD_NEXT, __func__);
  rmw_ret_t ret = ((functionT)orig_func)(publisher, serialized_message, allocation);
  return ret;
```